tests/unittests/GraphOptzTest.cpp

```diff
@@ -283,6 +283,9 @@ TEST_F(GraphOptz, optimizeBatchNormAfterConvFP16) {
   ASSERT_EQ(newCV->getNumUsers(), 1);
   Node *save = newCV->getUsers().begin()->getUser();
   EXPECT_TRUE(llvm::isa<SaveNode>(save));
+
+  optimizedF_ = optimizeFunction(F_);
```

Comment:
The numerical equivalence check feeds the same input to both `F_` and `optimizedF_`. `F_` at this point is already optimized due to line 277, so the numerical equivalence check is not useful. You should not optimize `F_` - this is where the `optimizeFunction` helper comes into play. It clones the function first, and then optimizes the clone.